server/app/forms/translation/ProgramTranslationForm.java

```diff
@@ -1,40 +1,153 @@
-  private String displayDescription;
+  private ProgramTranslationForm(DynamicForm form, int maxStatusTranslations) {
+    this.form = checkNotNull(form);
+    this.maxStatusTranslations = maxStatusTranslations;
+  }
 
-  public ProgramTranslationForm() {
-    this.displayName = "";
-    this.displayDescription = "";
+  private Optional<String> getStringFormField(String fieldName) {
+    return Optional.ofNullable(form.rawData().get(fieldName));
   }
 
-  /** Initializes a ProgramStatusesForm from the given status object. */
-  public static ProgramTranslationForm fromProgram(ProgramDefinition program, Locale locale) {
-    ProgramTranslationForm form = new ProgramTranslationForm();
-    form.setDisplayName(program.localizedName().maybeGet(locale).orElse(""));
-    form.setDisplayDescription(program.localizedDescription().maybeGet(locale).orElse(""));
-    return form;
+  public LocalizationUpdate getUpdateData() {
+    return LocalizationUpdate.builder()
+        .setLocalizedDisplayName(getStringFormField(DISPLAY_NAME_FORM_NAME).orElse(""))
+        .setLocalizedDisplayDescription(
+            getStringFormField(DISPLAY_DESCRIPTION_FORM_NAME).orElse(""))
+        .setStatuses(getStatusUpdates())
+        .build();
   }
 
-  public String getDisplayName() {
-    return displayName;
+  private ImmutableList<LocalizationUpdate.StatusUpdate> getStatusUpdates() {
+    return IntStream.range(0, maxStatusTranslations)
+        .boxed()
+        .map(
+            i -> {
+              Optional<String> maybeConfiguredStatusText =
+                  getStringFormField(configuredStatusFieldName(i));
+              Optional<String> maybeStatusText = getStringFormField(statusTextFieldName(i));
+              Optional<String> maybeEmail = getStringFormField(statusEmailFieldName(i));
+              if (maybeConfiguredStatusText.isEmpty() || maybeStatusText.isEmpty()) {
+                return Optional.empty();
+              }
+
+              LocalizationUpdate.StatusUpdate.Builder resultBuilder =
+                  LocalizationUpdate.StatusUpdate.builder()
+                      .setConfiguredStatusText(maybeConfiguredStatusText.get());
+              if (!maybeStatusText.get().isEmpty()) {
+                resultBuilder.setLocalizedStatusText(maybeStatusText);
+              }
+              if (maybeEmail.isPresent() && !maybeEmail.get().isEmpty()) {
+                resultBuilder.setLocalizedEmailBody(maybeEmail);
+              }
+              return Optional.of(resultBuilder.build());
+            })
+        .filter(maybeStatusUpdate -> maybeStatusUpdate.isPresent())
+        .map(maybeStatusUpdate -> (LocalizationUpdate.StatusUpdate) maybeStatusUpdate.get())
+        .collect(ImmutableList.toImmutableList());
   }
 
-  public void setDisplayName(String displayName) {
-    this.displayName = displayName;
+  public static String configuredStatusFieldName(int index) {
+    return String.format("status-%d-configured", index);
```

Comment:
suggestion: these 3 name patterns are effectively a triple but they don't read as such or discernibly to me at least.

If I understand their relationship, here's a stab at more aligned names.
configured -> configured-status
statusText ->localized-status
email -> localized-email